src/platform/plugins/shared/data/common/search/expressions/esql.ts

```diff
@@ -391,7 +396,14 @@ export const getEsqlFn = ({ getStartDependencies }: EsqlFnArguments) => {
           }
           const columnNames = updatedWithVariablesColumns?.map(({ name }) => name);
 
-          const rows = body.values.map((row) => zipObject(columnNames, row));
+          const rows = body.values.map((row) =>
+            columnNames.reduce<Record<string, unknown>>((ret, c, i) => {
+              const v = row[i];
+              // replace every `null` value with the `__missing__` to handle correctly the formatting
+              ret[c] = v == null ? '__missing__' : v;
```

Comment:
This might unexpectedly affect other logic in Kibana so I would not recommend it.
For example, "Hide null fields" switch is not working correctly on this branch:

<img width="1320" alt="Screenshot 2025-06-20 at 13 41 09" src="https://github.com/user-attachments/assets/8f9bb5d8-da55-4068-8ff2-aea1d958c8ec" />

Instead of overriding records data, a safer way would be to move this logic to the formatter code (if it's still relevant) somewhere around these lines: 
```
if (value == null) {
      return asPrettyString(value, options);
    }
```